utils/fork-tree/src/lib.rs

```diff
@@ -696,43 +733,56 @@ mod node_implementation {
 
 		pub fn import<F, E: std::error::Error>(
```

Comment:
>In the meantime I also spotted a bug in the original implementation, the tree would only get rebalanced when we pushed something to the root (the code above doesn't have that bug but that will probably affect the test results, with the self.rebalance() inside the None branch it should pass all tests). The reason import wasn't written like this in the first place is because find_node_where_mut was added after import.

Yep, so with this change, the tests are failing. Should I be changing the test values to match what now happens?

Reply:
Yeah I think so, the `test_fork_tree` will end up with a different layout when we build it which will eventually lead to different values when iterating it. Could you please just make sure that if `self.rebalance()` is only done inside the `None` branch, the tests pass without any changes? This would make us more confident that the change in tests is only caused by this fix.

Reply:
Yep, the tests pass if you do that.